infra/index.ts

```diff
@@ -0,0 +1,37 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+
+import exampleHandler from "../src/lambda";
+
+type Stage = "dev" | "beta" | "prod";
+
+// Read the stage from the STAGE environment variable
+const STAGE = process.env.STAGE as Stage ?? "dev";
+const SERVICE_NAME = "service-template";
+
+const createResourceName = (name: string) => `${STAGE}-${SERVICE_NAME}-${name}`;
+
+// Create an example S3 bucket
+const bucket = new aws.s3.Bucket(createResourceName("bucket"));
+
+// Create a role for the Lambda with no permissions, since the default role that
+// CallbackFunction creates is very unrestricted
+const lambdaRole = new aws.iam.Role(createResourceName("lambda-role"), {
+  assumeRolePolicy: aws.iam.assumeRolePolicyForPrincipal({ Service: "lambda.amazonaws.com" }),
+});
+
+// Create an example Lambda Function. This uses Pulumi Function Serialization. If we
+// run into issues with this we can use `aws.lambda.Function` instead, bundle the code
+// into a zip ourselves, and provide a path to it. See:
+// https://www.pulumi.com/docs/concepts/function-serialization/
+// https://www.pulumi.com/registry/packages/aws/api-docs/lambda/function/
+const lambda = new aws.lambda.CallbackFunction(createResourceName("lambda"), {
+  callback: exampleHandler,
+  runtime: aws.lambda.Runtime.NodeJS20dX,
+  role: lambdaRole.arn,
+});
+
+// Export the name of the bucket and the lambda arn
+export const bucketName = bucket.id;
```

Comment:
nit: should we return this as an object? so we can group values into nested objects if we want?

Reply:
We can return objects if we want, but wrapping all outputs in one object (even just a default export) makes it a bit more ugly to import them on other stacks as we'd have to do: `stackRef.getOutput("default").bucketName` instead of `stackRef.getOutput("bucketName")`. So I think let's save using objects until we want to group some values together.